dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -980,64 +981,85 @@ def _DownloadResults(self,
       results.append(data_frame)
     return results
 
+  def _ProcessQuery(
+      self,
+      hostname: str,
+      osquery_container: containers.OsqueryQuery
+  ) -> None:
+    """Processes an osquery flow on a GRR client.
+
+    Args:
+      hostname (str): the GRR client hostname.
+      osquery_container (containers.OsqueryQuery): the osquery.
+    """
+    client = self._GetClientBySelector(hostname)
```

Comment:
Can we extract this call outside the threadpoolexecutor map call? Having it here means that we fetch the GRR client every time we want to launch a flow, and we only need it once (we can pass a GRR client object here instead of a `hostname`)

Reply:
Done